sw/airborne/modules/computer_vision/video_usb_logger.c

```diff
@@ -28,35 +28,118 @@
 #include <stdio.h>
 #include "state.h"
 #include "viewvideo.h"
+#include "cv.h"
+#include <unistd.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
 
 /** Set the default File logger path to the USB drive */
 #ifndef VIDEO_USB_LOGGER_PATH
 #define VIDEO_USB_LOGGER_PATH /data/video/usb
 #endif
 
+#ifndef VIDEO_LOG_WIDTH
+#define VIDEO_LOG_WIDTH 272
+#endif
+
+#ifndef VIDEO_LOG_HEIGHT
+#define VIDEO_LOG_HEIGHT 272
+#endif
+
 /** The file pointer */
 static FILE *video_usb_logger = NULL;
+struct image_t img_jpeg;
+char foldername[512];
+int shotNumber = 0;
+
+void save_shot(struct image_t *img, struct image_t *img_jpeg)
+{
+
+  // Search for a file where we can write to
+  char save_name[128];
+
+  sprintf(save_name, "%s/img_%05d.jpg", foldername, shotNumber);
+
+  shotNumber++;
+  // Check if file exists or not
+  if (access(save_name, F_OK) == -1) {
+
+    // Create a high quality image (99% JPEG encoded)
+    jpeg_encode_image(img, img_jpeg, 99, TRUE);
+
+#if JPEG_WITH_EXIF_HEADER
```

Comment:
this is an option that you should define in your module/airframe file?
Maybe it should be specific to this module?